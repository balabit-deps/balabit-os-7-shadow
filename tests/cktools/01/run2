#!/bin/sh

set -e

cd $(dirname $0)

# Rational:
# Test that useradd can add an user and userdel removes it.

save()
{
	[ ! -d tmp ] && mkdir tmp
	for i in passwd group shadow gshadow
	do
		[ -f /etc/$i  ] && cp /etc/$i  tmp/$i
	done

	true
}

restore()
{
	rm -f tmp/err tmp/out
	for i in passwd group shadow gshadow
	do
		[ -f tmp/$i  ] && cp tmp/$i  /etc/$i  && rm tmp/$i
	done
	rmdir tmp
}

save

# restore the files on exit
trap 'restore' 0

for i in passwd group shadow gshadow
do
	cp data/$i /etc
done

lines_passwd=$(wc -l /etc/passwd | cut -f1 -d" ")
lines_shadow=$(wc -l /etc/shadow | cut -f1 -d" ")
lines_group=$(wc -l /etc/group | cut -f1 -d" ")
lines_gshadow=$(wc -l /etc/gshadow | cut -f1 -d" ")

echo -n "Add an user without an entry in shadow "
echo "test:x:10002:10002::/tmp:/bin/false" >> /etc/passwd
echo "OK"

echo "Check that pwck detects it "
pwck -r > tmp/out 2> tmp/err || true
diff -au data/run2.out tmp/out
diff -au data/run2.err tmp/err
echo "  OK"
echo "grpck accepts valid password file "
msg=$(grpck -r)
test "$msg" = ""
echo "  OK"

#echo -n "Make sure pwck can fix it "
#pwcd
#echo "OK"
