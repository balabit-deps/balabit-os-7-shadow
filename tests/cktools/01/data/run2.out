user 'lp': directory '/var/spool/lpd' does not exist
user 'news': directory '/var/spool/news' does not exist
user 'uucp': directory '/var/spool/uucp' does not exist
user 'www-data': directory '/var/www' does not exist
user 'list': directory '/var/list' does not exist
user 'irc': directory '/var/run/ircd' does not exist
user 'gnats': directory '/var/lib/gnats' does not exist
user 'nobody': directory '/nonexistent' does not exist
user 'Debian-exim': directory '/var/spool/exim4' does not exist
user 'test': no group 10002
no matching password file entry in /etc/shadow
add user 'test' in /etc/shadow? No
pwck: no changes
